### ISSUE TITLE
Opencart 3.0

### ISSUE THREAD

**Issue by prhost** [copona/copona#45]
Let's wait for Opencart 3.0? What will it be like when he leaves? What is Copona's plan?

**Comment by arnisjuraga** [comment]
Does OC 3.0 is available for preview? 
Does anyone has any release notes for OC 3.0?
Isn't it the latest source available from github.com/opencart/opencart ?

I would say, that Copona's plan is to "make it's own way" , so - if there will be something usable/implementation worth for Copona - we can use it.

**Comment by prhost** [comment]
Opencart 3.0 is in the dev branch, it is not scheduled to launch as usual, but a lot has already been done, if it is interest to get Copona if it is useful, so my question, are we going to wait or can we get it?

sorry, I guess I was not very expressive hehe

**Comment by arnisjuraga** [comment]
I will review it.
Until now, the only "huge" change I have been aware, is a Twig engine. This is something we definitely would want to implement! 
But from OC experience, I wouldn't be blind and hope, that "this time" OC 3.0 release at that moment will be "Stable" :)
and it will take 1 more year, to make it such.
So, I would be against it at the moment.

**Comment by prhost** [comment]
Yes I believe in you hehe, something else, in the time of accepting the twig, I did a very good pull request to improve the implementation of twig, it was practically modular, it was very good, but Daniel I think it was too much for his head Accept and refused hehehe. I also have a class extension of twig only with functions and filters suitable for Opencart. I use it and I can easily implement it for us.

**Comment by prhost** [comment]
What are the next steps of the copona? I'm touching my idea but I think it's going to take a while

**Comment by arnisjuraga** [comment]
We are launching few projects in next month, so there will be rapid code changes.
For now - we will not make anything "crazy" because Copona is going to be live for few webs.

**Comment by arnisjuraga** [comment]
Closing for now, will look back, although - may there will be no need for merging a lot.

**Comment by prhost** [comment]
I wonder if we will use opencart 3.0 or from the current version of copona, will we go our way? For after the release of version 3, I believe that Daniel will no longer support old versions, as he always did, and gradually we will lose link with opencart.

**Comment by arnisjuraga** [comment]
It's a bit a discussion.
At first - I suggest looking at OC3 "after some months or a year", because of the roadmap and the way how the OC is developed. I mean - it will take the time to get stable OC3. 

At second - we need to decide, what exactly do we need from OC3 branch? 
As we know - we are not benefitting from extension installer or extension engine at all, as we do not plan to support it. But we can integrate it when it will be stable in OC3 anyway, for the sake of compatibility.

Twig, on the other hand, can be very useful. And we need to decide on implementing it.

But what about all the other "changes"? If there anything at all, we can benefit from and it worth it - Copona can take the advantage.

**Comment by prhost** [comment]
Also does not find interesting the installer extension, opencart is not modular, and this may have fail and cause to lose codes, than actually facilitate the use.
Twig is very good and important, but the way they implemented it, it would never do so, I once sent to opencart, an implementation of twig, which left the modular template hander, but he declined.

Has some interesting features like theme editor, dynamic menu, dynamic dashboard, but I think we would do better than it was done in opencart hehe, but copying these features would be faster, just by adapting them.

The opencart theme of both catalog and admin, which was converted to twig, would save us a lot of work as well, we can copy whatever is compatible with our version.

**Comment by arnisjuraga** [comment]
Hey, guys, what's your opinion on OC3 version integration?

**Comment by prhost** [comment]
What to take advantage of version 3 is some things that I foresaw to do in the future as dynamic dashboard, dynamic menu and dynamic translation, of course there would be a lot of change in them, also would not integrate with the twig as Daniel did, but it would already be a Step faster because it is easier to modify and improve than to do everything from scratch. It also has the issue of improvements and bugfix that is always important, just throw away the part of extension that installs via opencart API, it downloads opencart site extensions, unpacks and plays in folders, this runs a serious risk of overwriting something, Then it would not. The problem is that to pull everything ta, it would have to be manual and slow, a lot of things will break, because they are big versions of difference. I can only pull with cherrypick what we want, with all that I said, also to undecided to pull or not hehe.

**Comment by arnisjuraga** [comment]
Hmm, dynamic menu and dynamoc translations - have you looked around how its implemented?

**Comment by prhost** [comment]
I looked over, both are via database, I think bad, especially the transaltion, but it's not so bad for not to exist, this was all done with the purpose of the panel user not tampering with the code, he's creating opencart for Being worked on the cloud of his, also he put theme editor via panel, which also has the purpose of not letting the client directly access the php code as the controllers and model.

**Comment by arnisjuraga** [comment]
You are right, it more looks to be dedicated to end user, not dev framework

**Comment by prhost** [comment]
Has also been modified many things in the copona, this will conflict a lot, among the employees, there is a frontend? We could redraw the admin equal the arastta, or use these panel ready like https://github.com/almasaeed2010/AdminLTE
